src/NUnitFramework/framework/Is.cs

```diff
@@ -141,6 +141,15 @@ public abstract class Is
 
         #region EqualTo
 
+        /// <summary>
+        /// Returns a constraint that tests two items for equality
+        /// </summary>
+        public static EqualConstraint<T> EqualTo<T>(T expected)
```

Comment:
Something for consideration. Adding this overload will cause at least some call sites previously resolving to the `object` overload to now resolve here. The difference in return type between these methods _could_ then be breaking unless our new `EqualConstraint<T>` were to subclass `EqualConstraint` itself in the short term.

Reply:
Or perhaps the reverse
```
EqualConstraint : EqualConstraint<object>
```

Reply:
Yes, I had thought of that as well.

Reply:
Thanks! With "deep coding" time harder to come by at the moment, these comments were a way of documenting where I had to leave off for the next time myself or someone else continues to chip away. 

Glad to hear we've been thinking about this the same way!